src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import duke.command.Command;
+import javafx.scene.layout.VBox;
+
+public class Duke {
+
+    public static final String DATA_FILE_TASKS = "./data/duke.txt";
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Creates an instance of Duke, setting up the UI, storage, and task list.
+     */
+    public Duke() {
+        ui = new SystemUi();
+        storage = new Storage(DATA_FILE_TASKS);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void setupGuiUi(VBox dialogContainer) {
+        ui = new GuiUi(dialogContainer);
+    }
+
+    /**
+     * Runs the instance of Duke with the main loop.
+     */
+    public void run() {
+        ui.showWelcome();
+        ui.showLine();
+        boolean isExit = false;
+        while (!isExit) {
+            String fullCommand = ui.readCommand();
+            isExit = respond(fullCommand);
+        }
+        try {
+            storage.save(tasks.dump());
+        } catch (DukeException e) {
+            ui.showSavingError();
+        }
+    }
+
+    /**
+     * This is the main method and entry point for the Duke program.
+     * @param args Command line arguments
+     */
+    public static void main(String[] args) {
+        new Duke().run();
+    }
+
+    /**
+     * Returns whether the program should exit
+     * @param input Input string
+     * @return Boolean indicating of program should exit
+     */
+    public boolean respond(String input) {
```

Comment:
This is a very good method to use. Because the website itself only show the hint for the front part and you adding this is a very smart move :)

Reply:
This seems to be a good method to use for the CLI. Will it be applicable as well for the GUI?